[processor/filterprocessor] Broken `TestLoadingConfigOTTL` tests

Part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/17037

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3706734880/jobs/6282715360

```
make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor'
go test -race -timeout 300s --tags="" ./...
--- FAIL: TestLoadingConfigOTTL (0.04s)
    --- FAIL: TestLoadingConfigOTTL/filter/bad_syntax_span (0.00s)
        config_test.go:9[38](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3706734880/jobs/6282715360#step:9:39): 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor/config_test.go:938
            	Error:      	Error message not equal:
            	            	expected: "1:24: unexpected token \"[\" (expected <opcomparison> Value)"
            	            	actual  : "1:24: unexpected token \"[\" (expected <opcomparison> Value); unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"
            	Test:       	TestLoadingConfigOTTL/filter/bad_syntax_span
    --- FAIL: TestLoadingConfigOTTL/filter/bad_syntax_spanevent (0.00s)
        config_test.go:938: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor/config_test.go:938
            	Error:      	Error message not equal:
            	            	expected: "1:24: unexpected token \"[\" (expected <opcomparison> Value)"
            	            	actual  : "1:24: unexpected token \"[\" (expected <opcomparison> Value); unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"
            	Test:       	TestLoadingConfigOTTL/filter/bad_syntax_spanevent
    --- FAIL: TestLoadingConfigOTTL/filter/bad_syntax_metric (0.00s)
        config_test.go:938: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor/config_test.go:938
            	Error:      	Error message not equal:
            	            	expected: "1:33: unexpected token \"[\" (expected <opcomparison> Value)"
            	            	actual  : "1:33: unexpected token \"[\" (expected <opcomparison> Value); unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"
            	Test:       	TestLoadingConfigOTTL/filter/bad_syntax_metric
    --- FAIL: TestLoadingConfigOTTL/filter/bad_syntax_datapoint (0.00s)
        config_test.go:938: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor/config_test.go:938
            	Error:      	Error message not equal:
            	            	expected: "1:24: unexpected token \"[\" (expected <opcomparison> Value)"
            	            	actual  : "1:24: unexpected token \"[\" (expected <opcomparison> Value); unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"
            	Test:       	TestLoadingConfigOTTL/filter/bad_syntax_datapoint
    --- FAIL: TestLoadingConfigOTTL/filter/bad_syntax_log (0.00s)
        config_test.go:938: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/processor/filterprocessor/config_test.go:938
            	Error:      	Error message not equal:
            	            	expected: "1:24: unexpected token \"[\" (expected <opcomparison> Value)"
            	            	actual  : "1:24: unexpected token \"[\" (expected <opcomparison> Value); unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"
            	Test:       	TestLoadingConfigOTTL/filter/bad_syntax_log
```

--- a/pkg/ottl/parser.go
+++ b/pkg/ottl/parser.go
@@ -103,7 +103,7 @@ func parseStatement(raw string) (*parsedStatement, error) {
 	parsed, err := parser.ParseString("", raw)
 
 	if err != nil {
-		return nil, multierr.Append(err, fmt.Errorf("unable to parse OTTL statement, ensure the statement's syntax is correct; common mistakes include missing parentheses, missing double quotes and incorrect function name case"))
+		return nil, fmt.Errorf("unable to parse OTTL statement: %w", err)
 	}
 	err = parsed.checkForCustomError()
 	if err != nil {

--- a/processor/filterprocessor/config_test.go
+++ b/processor/filterprocessor/config_test.go
@@ -905,23 +905,23 @@ func TestLoadingConfigOTTL(t *testing.T) {
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_span"),
-			errorMessage: "1:24: unexpected token \"[\" (expected <opcomparison> Value)",
+			errorMessage: "unable to parse OTTL statement: 1:24: unexpected token \"[\" (expected <opcomparison> Value)",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_spanevent"),
-			errorMessage: "1:24: unexpected token \"[\" (expected <opcomparison> Value)",
+			errorMessage: "unable to parse OTTL statement: 1:24: unexpected token \"[\" (expected <opcomparison> Value)",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_metric"),
-			errorMessage: "1:33: unexpected token \"[\" (expected <opcomparison> Value)",
+			errorMessage: "unable to parse OTTL statement: 1:33: unexpected token \"[\" (expected <opcomparison> Value)",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_datapoint"),
-			errorMessage: "1:24: unexpected token \"[\" (expected <opcomparison> Value)",
+			errorMessage: "unable to parse OTTL statement: 1:24: unexpected token \"[\" (expected <opcomparison> Value)",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_log"),
-			errorMessage: "1:24: unexpected token \"[\" (expected <opcomparison> Value)",
+			errorMessage: "unable to parse OTTL statement: 1:24: unexpected token \"[\" (expected <opcomparison> Value)",
 		},
 	}
 

--- a/processor/transformprocessor/config_test.go
+++ b/processor/transformprocessor/config_test.go
@@ -131,7 +131,7 @@ func TestLoadConfig(t *testing.T) {
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_trace"),
-			errorMessage: "1:18: unexpected token \"where\" (expected \")\")",
+			errorMessage: "unable to parse OTTL statement: 1:18: unexpected token \"where\" (expected \")\")",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "unknown_function_trace"),
@@ -139,7 +139,7 @@ func TestLoadConfig(t *testing.T) {
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_metric"),
-			errorMessage: "1:18: unexpected token \"where\" (expected \")\")",
+			errorMessage: "unable to parse OTTL statement: 1:18: unexpected token \"where\" (expected \")\")",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "unknown_function_metric"),
@@ -147,7 +147,7 @@ func TestLoadConfig(t *testing.T) {
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "bad_syntax_log"),
-			errorMessage: "1:18: unexpected token \"where\" (expected \")\")",
+			errorMessage: "unable to parse OTTL statement: 1:18: unexpected token \"where\" (expected \")\")",
 		},
 		{
 			id:           component.NewIDWithName(typeStr, "unknown_function_log"),

